components/mdns/mdns_networking_socket.c

```diff
@@ -346,17 +352,9 @@ static bool create_pcb(mdns_if_t tcpip_if, mdns_ip_protocol_t ip_protocol)
     }
     int sock = s_interfaces[tcpip_if].sock;
     esp_netif_t *netif = _mdns_get_esp_netif(tcpip_if);
-    if (sock >= 0) {
-        mdns_ip_protocol_t other_ip_proto = ip_protocol == MDNS_IP_PROTOCOL_V4 ? MDNS_IP_PROTOCOL_V6 : MDNS_IP_PROTOCOL_V4;
-        int err = join_mdns_multicast_group(sock, netif, other_ip_proto);
-        if (err < 0) {
-            ESP_LOGE(TAG, "Failed to add ipv6 multicast group for protocol %d", ip_protocol);
-            return false;
-        }
-        s_interfaces[tcpip_if].proto |= (other_ip_proto == MDNS_IP_PROTOCOL_V4 ? PROTO_IPV4 : PROTO_IPV6);
```

Comment:
I wonder why this was removed? The idea is that we already have a socket supporting one IP protocol (say IPv4) and someone calls `create_pcb(same_netif, IPv6)`. Then we'll never join the IPv6 multicast group, correct?

Reply:
The `create_pcb(some_netif, IPv6)` will still call `join_mdns_multicast_group(sock, netif, IPv6);` in the following logic.

But if we keep the original logic, When we already have a socket supporting IPv4, and then someone calls `create_pcb(some_netif, IPv4)`, the netif will never join IPv4 multicast group.

Reply:
Oh, I see. so you actually always join the multicast group (again).
This would (probably?) work as we guard this in `mdns.c` using `mdns_is_netif_ready()` API.